### ISSUE TITLE
Support open/close pairs

### ISSUE THREAD

**Issue by GlenPerkins** [BriteSnow/vscode-toggle-quotes#15]
I'd love to be able to use this to make smart/curly quotes an option in text files. Straight quotes convert to curly and vice-versa.

This would of course require more logic, because the char inserted at the beginning would differ from the one at the end, but that additional logic would enable such things as <>, {}, [], and ().

And if your code doesn't require single-char strings, it would also enable more open/close pairs that the user could specify per filetype: {{ }}, <b> </b>, etc.

**Comment by jeremychone** [comment]
Interesting feature. That would be on a different key map, like a "cmd + {" kind of thing, right?

**Comment by GlenPerkins** [comment]
Yes, maybe cycling through quotes with one command and cycling through <{[( with another. 

BUT, I'm probably combining too many things. My primary interest (in addition to writing code) is to be able to use VS Code to write natural language, such as a blog post, user manual, article, book, etc. These need to use proper ("smart/curly") quotes and apostrophes, which you can get from any word processor, but then you have to copy back and forth to keep it in your git repo...a mess. So what I'm looking for is an easy way to use smart ("curly") quotes easily within VS Code. Ideally, it would be smart enough to do the whole job, toggling between:
"That's a 'problem', I'd say." <> ”That‘s a ’problem‘, I‘d say.”
with a selection and keystroke.

This isn't your responsibility, of course, but a simpler tool that just cycled (user-specified) matched-pair delimiters, ex:
'foo'  -> ‘foo’ -> "foo" -> “foo” -> `foo` -> (repeat)
would be very convenient, and you are already pretty close to doing it. It would be especially nice if it had an extra bit of intelligence so that it treated [alpha]['|’][alpha] as an apostrophe, NOT a delimiter, and simply converted an apostrophe between delimiters to curly when the outside delimiters were curly and straight otherwise. There will always be uncommon cases where, for example, you cut off the front of a word and need an apostrophe (ex: ’tis ’round ’bout 4am), but those edge cases are where you ignore the convenient tool and just edit manually.
